[{"count": 0, "tags": [], "creator": "mcurry@more.net", "attachment_id": null, "id": 4310, "time": "2001-08-06T14:52:42Z", "bug_id": 3009, "creation_time": "2001-08-06T14:52:42Z", "is_private": false, "text": "When attempting to install a web app war file with the manager web application,\nit reports...\n\nFAIL - Encountered exception java.io.IOException: The system cannot find the\npath specified\n\n...The directory for the web app is created, but the directory is empty.  This\noccurs for any web app which is trying to be installed.\n\nThis worked previously under beta 5.\nSecurity is -not- in use.\nSystem is a: PentiumIII PC running Windows2000.\nUsing: jdk 1.3.1 ( c:\\jdk1.3.1\\jre\\bin\\hotspot\\jvm.dll ).\nThis has been duplicated on other win2000 machines.\n\n-- Copy of the log...\n\n2001-08-06 15:56:56 StandardWrapper[/manager:Manager]: Loading container servlet\nManager\n2001-08-06 15:56:56 Manager: init\n2001-08-06 15:56:56 Manager: init: Associated with Deployer 'localhost'\n2001-08-06 15:56:56 Manager: install: Installing web application at\n'/application-doc' from\n'jar:http://kinetic.more.net/download/taglib/application-doc.war!/'\n2001-08-06 15:56:56 StandardHost[localhost]: Installing web application at\ncontext path /application-doc from URL\njar:http://kinetic.more.net/download/taglib/application-doc.war!/\n2001-08-06 15:56:56 WebappLoader[/application-doc]: Deploying class repositories\nto work directory C:\\tomcat4\\work\\localhost\\application-doc\n2001-08-06 15:56:56 StandardManager[/application-doc]: Seeding random number\ngenerator class java.security.SecureRandom\n2001-08-06 15:56:56 StandardManager[/application-doc]: Seeding of random number\ngenerator has been completed\n2001-08-06 15:56:56 ContextConfig[/application-doc]: Missing application\nweb.xml, using defaults only\n2001-08-06 15:56:56 StandardWrapper[/application-doc:default]: Loading container\nservlet default\n2001-08-06 15:56:56 default: init\n2001-08-06 15:56:56 StandardWrapper[/application-doc:invoker]: Loading container\nservlet invoker\n2001-08-06 15:56:56 invoker: init\n2001-08-06 15:56:56 StandardWrapper[/application-doc:jsp]: Using Jasper\nclassloader for servlet jsp\n2001-08-06 15:56:56 jsp: init\n2001-08-06 15:56:56 StandardWrapper[/application-doc:ssi]: Loading container\nservlet ssi\n2001-08-06 15:56:56 ssi: init\n2001-08-06 15:56:56 Internal Error: File /WEB-INF/web.xml not found\n2001-08-06 15:56:56 \n2001-08-06 15:57:15 Manager: install: Installing web application at\n'/application-doc' from\n'jar:http://kinetic.more.net/download/taglib/application-doc.war!/'\n2001-08-06 15:57:44 StandardHost[localhost]: Removing web application at context\npath /examples\n2001-08-06 15:57:44 StandardHost[localhost]: Removing web application at context\npath /webdav\n2001-08-06 15:57:44 ssi: destroy\n2001-08-06 15:57:44 StandardHost[localhost]: Removing web application at context\npath /manager\n2001-08-06 15:57:44 ssi: destroy"}, {"count": 1, "tags": [], "creator": "remm@apache.org", "attachment_id": null, "id": 4397, "time": "2001-08-08T18:09:45Z", "bug_id": 3009, "creation_time": "2001-08-08T18:09:45Z", "is_private": false, "text": "I've had trouble reproducing the bug, but I've experienced some weird behavior \nat some point (which I haven't been able to reproduce afterwards). I also don't \nbelieve any of that code changed between b5 and now.\n\n2001-08-08 17:52:42 Manager: install: Installing web application at '/foo3' \nfrom 'jar:http://127.0.0.1/foo2.war!/'\n2001-08-08 17:52:42 StandardHost[localhost]: Installing web application at \ncontext path /foo3 from URL jar:http://127.0.0.1/foo2.war!/\n2001-08-08 17:52:43 WebappLoader[/foo3]: Deploying class repositories to work \ndirectory D:\\cvs\\jakarta-tomcat-4.0\\dist\\work\\localhost\\foo3\n......\n\nThe foo3 directory correctly contains the unpacked contents of the WAR file, \nand the context is running fine."}, {"count": 2, "tags": [], "creator": "mcurry@more.net", "is_private": false, "id": 4462, "attachment_id": null, "bug_id": 3009, "creation_time": "2001-08-10T12:30:36Z", "time": "2001-08-10T12:30:36Z", "text": "We were able to reproduce it on another Windows2000 machine.  Under unix, it \nworks fine.\n\nI tried it again after cleaning out the logs.  I used security manager this \ntime, but my catalina.policy is set to grant full permission to everything and \nused \"-Djava.security.debug=access,failure\" when I started Tomcat.  \n\nAgain, I got the \"FAIL - Encountered exception java.io.IOException: The system \ncannot find the path specified\" reported back in my web browser.  Checking the \ndebug output, it is a straight line of \"access allowed\" all the way down.\n\nWould this be of help?  This is the relevant portion of my \nlocalhost_log.2001-08-10.txt file...\n-----------------------------------------\n2001-08-10 14:09:53 StandardWrapper[/manager:Manager]: Loading container servlet \nManager\n2001-08-10 14:09:53 Manager: init\n2001-08-10 14:09:53 Manager: init: Associated with Deployer 'localhost'\n2001-08-10 14:09:53 Manager: install: Installing web application at \n'/datetime-doc' from \n'jar:http://kinetic.more.net/download/taglib/datetime-doc.war!/'\n2001-08-10 14:09:53 StandardHost[localhost]: Installing web application at \ncontext path /datetime-doc from URL \njar:http://kinetic.more.net/download/taglib/datetime-doc.war!/\n2001-08-10 14:09:53 ManagerServlet.install[/datetime-doc]\njava.io.IOException: The system cannot find the path specified\n\tat java.io.Win32FileSystem.createFileExclusively(Native Method)\n\tat java.io.File.checkAndCreate(File.java:1162)\n\tat java.io.File.createTempFile(File.java:1247)\n\tat java.io.File.createTempFile(File.java:1284)\n\tat sun.net.www.protocol.jar.URLJarFile$1.run(URLJarFile.java:143)\n\tat java.security.AccessController.doPrivileged(Native Method)\n\tat sun.net.www.protocol.jar.URLJarFile.retrieve(URLJarFile.java:138)\n\tat sun.net.www.protocol.jar.URLJarFile.getJarFile(URLJarFile.java:43)\n\tat sun.net.www.protocol.jar.JarFileFactory.get(JarFileFactory.java:61)\n\tat \nsun.net.www.protocol.jar.JarURLConnection.connect(JarURLConnection.java:88)\n\tat \nsun.net.www.protocol.jar.JarURLConnection.getJarFile(JarURLConnection.java:72)\n\tat org.apache.catalina.core.StandardHost.expand(StandardHost.java:869)\n\tat org.apache.catalina.core.StandardHost.install(StandardHost.java:600)\n\tat \norg.apache.catalina.servlets.ManagerServlet.install(ManagerServlet.java:370)\n\tat \norg.apache.catalina.servlets.ManagerServlet.doGet(ManagerServlet.java:271)\n\tat javax.servlet.http.HttpServlet.service(HttpServlet.java:740)\n\tat javax.servlet.http.HttpServlet.service(HttpServlet.java:853)\n\tat \norg.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilt\nerChain.java:247)\n\tat \norg.apache.catalina.core.ApplicationFilterChain.access$0(ApplicationFilterChain.\njava:197)\n\tat \norg.apache.catalina.core.ApplicationFilterChain$1.run(ApplicationFilterChain.jav\na:176)\n\tat java.security.AccessController.doPrivileged(Native Method)\n\tat \norg.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.\njava:172)\n\tat \norg.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:2\n43)\n\tat \norg.apache.catalina.core.StandardPipeline.invokeNext(StandardPipeline.java:566)\n\tat \norg.apache.catalina.core.StandardPipeline.invoke(StandardPipeline.java:472)\n\tat org.apache.catalina.core.ContainerBase.invoke(ContainerBase.java:943)\n\tat \norg.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:2\n15)\n\tat \norg.apache.catalina.core.StandardPipeline.invokeNext(StandardPipeline.java:566)\n\tat \norg.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.jav\na:518)\n\tat \norg.apache.catalina.core.StandardPipeline.invokeNext(StandardPipeline.java:564)\n\tat \norg.apache.catalina.core.StandardPipeline.invoke(StandardPipeline.java:472)\n\tat org.apache.catalina.core.ContainerBase.invoke(ContainerBase.java:943)\n\tat \norg.apache.catalina.core.StandardContext.invoke(StandardContext.java:2314)\n\tat \norg.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:164)\n\tat \norg.apache.catalina.core.StandardPipeline.invokeNext(StandardPipeline.java:566)\n\tat \norg.apache.catalina.valves.AccessLogValve.invoke(AccessLogValve.java:446)\n\tat \norg.apache.catalina.core.StandardPipeline.invokeNext(StandardPipeline.java:564)\n\tat \norg.apache.catalina.core.StandardPipeline.invoke(StandardPipeline.java:472)\n\tat org.apache.catalina.core.ContainerBase.invoke(ContainerBase.java:943)\n\tat \norg.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:163\n)\n\tat \norg.apache.catalina.core.StandardPipeline.invokeNext(StandardPipeline.java:566)\n\tat \norg.apache.catalina.core.StandardPipeline.invoke(StandardPipeline.java:472)\n\tat org.apache.catalina.core.ContainerBase.invoke(ContainerBase.java:943)\n\tat \norg.apache.catalina.connector.http.HttpProcessor.process(HttpProcessor.java:995)\n\tat \norg.apache.catalina.connector.http.HttpProcessor.run(HttpProcessor.java:1088)\n\tat java.lang.Thread.run(Thread.java:484)\n\n2001-08-10 14:10:25 StandardHost[localhost]: Removing web application at context \npath /examples\n\n\n"}, {"count": 3, "tags": [], "creator": "remm@apache.org", "attachment_id": null, "is_private": false, "id": 5180, "time": "2001-09-06T16:26:02Z", "bug_id": 3009, "creation_time": "2001-09-06T16:26:02Z", "text": "Thanks for the additional details. The exception you get is quite strange, and\nI'll try to find the root cause of the problem."}, {"count": 4, "tags": [], "creator": "craig.mcclanahan@sun.com", "is_private": false, "id": 5380, "attachment_id": null, "bug_id": 3009, "creation_time": "2001-09-11T15:06:16Z", "time": "2001-09-11T15:06:16Z", "text": "Can not reproduce with current Tomcat code.  Could you try this again with\n4.0-rc1?  If you still have a problem, could you include a small WAR file that\nillustrates it for testing purposes?\n"}, {"count": 5, "tags": [], "creator": "mcurry@more.net", "is_private": false, "id": 6165, "attachment_id": null, "bug_id": 3009, "creation_time": "2001-09-28T08:42:40Z", "time": "2001-09-28T08:42:40Z", "text": "Okay, I've tried it again with the 4.0 release.\n\nI'm still getting the error under Windows (under unix it works fine).\n\nThe information you wanted plus some other stuff I hope will be helpful is\nhere...\n\nhttp://www.more.net/~mcurry/bug3009/\n\n"}, {"count": 6, "tags": [], "creator": "mcurry@more.net", "attachment_id": null, "id": 6169, "time": "2001-09-28T08:50:14Z", "bug_id": 3009, "creation_time": "2001-09-28T08:50:14Z", "is_private": false, "text": "p.s. - It creates the .\\webapps\\application-doc\\ directory, but does not create \nanything in it.  The application-doc directory ends up empty.\n"}]